code/game/turfs/simulated/floor/plating/asteroid.dm

```diff
@@ -9,18 +9,19 @@
 	icon = 'icons/turf/floors.dmi'
 	icon_state = "asteroid"
 	icon_plating = "asteroid"
+	postdig_icon_change = TRUE
 	var/environment_type = "asteroid"
 	var/turf_type = /turf/open/floor/plating/asteroid //Because caves do whacky shit to revert to normal
-	var/dug = 0       //0 = has not yet been dug, 1 = has already been dug
-	var/sand_type = /obj/item/ore/glass
 	var/floor_variance = 20 //probability floor has a different icon state
+	archdrops = list(/obj/item/ore/glass = 5)
 
 /turf/open/floor/plating/asteroid/Initialize()
 	var/proper_name = name
 	. = ..()
 	name = proper_name
 	if(prob(floor_variance))
 		icon_state = "[environment_type][rand(0,12)]"
+	AddComponent(/datum/component/archaeology, 100, archdrops)
```

Comment:
There should be a if(LAZYLEN(archdrops)) before this

Reply:
what does LAZYLEN do